packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -563,13 +563,18 @@ jobject convert_to_jvm_sync_error(JNIEnv* jenv, const realm_sync_error_t& error)
         realm_sync_error_user_info_t user_info = error.user_info_map[i];
         user_info_map->insert(std::make_pair(user_info.key, user_info.value));
     }
+
     if (error.user_info_length > 0) {
         auto original_it = user_info_map->find(error.c_original_file_path_key);
```

Comment:
In order to prevent this bug from happening again, we should probably check that this isn't user_info_map.end(). If this doesn't find anything, we will end up with similar memory corruption issues as we have now.